### PR TITLE
Fix Wazuh installation assistant AIO deployment in Fedora 36

### LINKED ISSUE #1702: Wazuh installation assistant fails to install an AIO deployment in Fedora 36 due to openssl not installed
|Wazuh version|Install type|Action performed|Platform|
|---|---|---|---|
| 4.3.4 | Installation assistant | Install AIO | Fedora 36 |

Investigating the issue https://github.com/wazuh/wazuh-packages/issues/1665, i have found that Fedora 36 use GLIBC 2.35, which uses the rseq syscall.

Trying to install an AIO deployment i found an openssl related error during the install process.

```
[root@fedora36 vagrant]# cat /etc/os-release 
NAME="Fedora Linux"
VERSION="36 (Thirty Six)"
ID=fedora
VERSION_ID=36
VERSION_CODENAME=""
PLATFORM_ID="platform:f36"
PRETTY_NAME="Fedora Linux 36 (Thirty Six)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:36"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f36/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=36
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=36
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
[root@fedora36 vagrant]#
```

----

### Openssl not found


<details><summary>Install output</summary>

```
[root@fedora36 vagrant]# curl -sO https://packages.wazuh.com/4.3/wazuh-install.sh && sudo bash ./wazuh-install.sh -a
28/06/2022 17:59:53 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.4
28/06/2022 17:59:53 INFO: Verbose logging redirected to /var/log/wazuh-install.log
28/06/2022 17:59:53 ERROR: The recommended systems are: Red Hat Enterprise Linux 7, 8; CentOS 7, 8; Amazon Linux 2; Ubuntu 16.04, 18.04, 20.04, 22.04. The current system doesn't match this list. Use -i|--ignore-check to skip this check.
[root@fedora36 vagrant]# curl -sO https://packages.wazuh.com/4.3/wazuh-install.sh && sudo bash ./wazuh-install.sh -a -i
28/06/2022 17:59:57 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.4
28/06/2022 17:59:57 INFO: Verbose logging redirected to /var/log/wazuh-install.log
28/06/2022 17:59:59 WARNING: Hardware and system checks ignored.
28/06/2022 18:00:01 INFO: Wazuh repository added.
28/06/2022 18:00:01 INFO: --- Configuration files ---
28/06/2022 18:00:01 INFO: Generating configuration files.
28/06/2022 18:00:01 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
28/06/2022 18:00:01 INFO: --- Wazuh indexer ---
28/06/2022 18:00:01 INFO: Starting Wazuh indexer installation.
28/06/2022 18:01:01 INFO: Wazuh indexer installation finished.
28/06/2022 18:01:01 ERROR: Tar file does not contain certificate for the node wazuh-indexer.
28/06/2022 18:01:01 INFO: --- Removing existing Wazuh installation ---
28/06/2022 18:01:01 INFO: Removing Wazuh indexer.
28/06/2022 18:01:02 INFO: Wazuh indexer removed.
28/06/2022 18:01:02 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
[root@fedora36 vagrant]#
```

</details>

<details><summary>Wazuh install log </summary>

```
  GNU nano 6.0                                                                                                                                                                                                                                                                                                                                                                                                                                                                  /var/log/wazuh-install.log                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
28/06/2022 17:59:57 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.4
28/06/2022 17:59:57 INFO: Verbose logging redirected to /var/log/wazuh-install.log
28/06/2022 17:59:59 WARNING: Hardware and system checks ignored.
[wazuh]
gpgcheck=1
gpgkey=https://packages.wazuh.com/key/GPG-KEY-WAZUH
enabled=1
name=EL-${releasever} - Wazuh
baseurl=https://packages.wazuh.com/4.x/yum/
protect=1
28/06/2022 18:00:01 INFO: Wazuh repository added.
28/06/2022 18:00:01 INFO: --- Configuration files ---
28/06/2022 18:00:01 INFO: Generating configuration files.
./wazuh-install.sh: line 2840: openssl: command not found
./wazuh-install.sh: line 2743: openssl: command not found
./wazuh-install.sh: line 2744: openssl: command not found
./wazuh-install.sh: line 2745: openssl: command not found
./wazuh-install.sh: line 2746: openssl: command not found
./wazuh-install.sh: line 2803: openssl: command not found
./wazuh-install.sh: line 2804: openssl: command not found
./wazuh-install.sh: line 2816: openssl: command not found
./wazuh-install.sh: line 2817: openssl: command not found
./wazuh-install.sh: line 2829: openssl: command not found
./wazuh-install.sh: line 2830: openssl: command not found
chmod: cannot access '/tmp/wazuh-certificates/*': No such file or directory
mv: cannot stat '/tmp/wazuh-certificates/*': No such file or directory
28/06/2022 18:00:01 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
28/06/2022 18:00:01 INFO: --- Wazuh indexer ---
28/06/2022 18:00:01 INFO: Starting Wazuh indexer installation.
Fedora 36 - x86_64                               49 kB/s |  20 kB     00:00
Fedora 36 openh264 (From Cisco) - x86_64        2.7 kB/s | 989  B     00:00
Fedora Modular 36 - x86_64                       97 kB/s |  20 kB     00:00
Fedora 36 - x86_64 - Updates                     67 kB/s |  19 kB     00:00
Fedora 36 - x86_64 - Updates                    5.9 MB/s |  11 MB     00:01
Fedora Modular 36 - x86_64 - Updates             91 kB/s |  19 kB     00:00
Fedora Modular 36 - x86_64 - Updates            406 kB/s | 673 kB     00:01
EL-36 - Wazuh                                    12 MB/s | 8.3 MB     00:00
Dependencies resolved.
================================================================================
 Package                        Arch        Version           Repository   Size
================================================================================
Installing:
 wazuh-indexer                  x86_64      4.3.4-1           wazuh	  361 M
Installing dependencies:
 chkconfig                      x86_64      1.19-2.fc36       fedora	  161 k
 initscripts                    x86_64      10.16-2.fc36      fedora	  213 k
 initscripts-rename-device	x86_64      10.16-2.fc36      fedora	   18 k

Transaction Summary
================================================================================
Install  4 Packages

Total download size: 362 M
Installed size: 616 M
Downloading Packages:
(1/4): chkconfig-1.19-2.fc36.x86_64.rpm         291 kB/s | 161 kB     00:00
(2/4): initscripts-10.16-2.fc36.x86_64.rpm	344 kB/s | 213 kB     00:00
(3/4): initscripts-rename-device-10.16-2.fc36.x  29 kB/s |  18 kB     00:00
(4/4): wazuh-indexer-4.3.4-1.x86_64.rpm          30 MB/s | 361 MB     00:11
--------------------------------------------------------------------------------
Total                                            28 MB/s | 362 MB     00:12
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1
  Installing	   : chkconfig-1.19-2.fc36.x86_64                           1/4
  Installing	   : initscripts-rename-device-10.16-2.fc36.x86_64          2/4
  Installing	   : initscripts-10.16-2.fc36.x86_64                        3/4
  Running scriptlet: initscripts-10.16-2.fc36.x86_64                        3/4
Created symlink /etc/systemd/system/sysinit.target.wants/import-state.service → /usr/lib/systemd/system/import-state.service.

  Running scriptlet: wazuh-indexer-4.3.4-1.x86_64                           4/4
  Installing	   : wazuh-indexer-4.3.4-1.x86_64                           4/4
  Running scriptlet: wazuh-indexer-4.3.4-1.x86_64                           4/4
Created opensearch keystore in /etc/wazuh-indexer/opensearch.keystore

  Verifying        : chkconfig-1.19-2.fc36.x86_64                           1/4
  Verifying        : initscripts-10.16-2.fc36.x86_64                        2/4
  Verifying        : initscripts-rename-device-10.16-2.fc36.x86_64          3/4
  Verifying        : wazuh-indexer-4.3.4-1.x86_64                           4/4

Installed:
  chkconfig-1.19-2.fc36.x86_64                  initscripts-10.16-2.fc36.x86_64
  initscripts-rename-device-10.16-2.fc36.x86_64 wazuh-indexer-4.3.4-1.x86_64

Complete!
28/06/2022 18:01:01 INFO: Wazuh indexer installation finished.
28/06/2022 18:01:01 ERROR: Tar file does not contain certificate for the node wazuh-indexer.
28/06/2022 18:01:01 INFO: --- Removing existing Wazuh installation ---
28/06/2022 18:01:01 INFO: Removing Wazuh indexer.
Dependencies resolved.
================================================================================
 Package                       Arch	  Version             Repository   Size
================================================================================
Removing:
 wazuh-indexer                 x86_64     4.3.4-1             @wazuh	  614 M
Removing unused dependencies:
 chkconfig                     x86_64     1.19-2.fc36         @fedora     764 k
 initscripts                   x86_64     10.16-2.fc36        @fedora     1.1 M
 initscripts-rename-device     x86_64     10.16-2.fc36        @fedora	   16 k

Transaction Summary
================================================================================
Remove  4 Packages

Freed space: 616 M
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1
  Running scriptlet: wazuh-indexer-4.3.4-1.x86_64                           1/4
Stopping wazuh-indexer service... OK

  Erasing          : wazuh-indexer-4.3.4-1.x86_64                           1/4
warning: /etc/wazuh-indexer/opensearch.yml saved as /etc/wazuh-indexer/opensearch.yml.rpmsave
warning: /etc/wazuh-indexer/jvm.options saved as /etc/wazuh-indexer/jvm.options.rpmsave

  Running scriptlet: wazuh-indexer-4.3.4-1.x86_64                           1/4
  Running scriptlet: initscripts-10.16-2.fc36.x86_64                        2/4
Removed /etc/systemd/system/sysinit.target.wants/import-state.service.

  Erasing          : initscripts-10.16-2.fc36.x86_64                        2/4
  Running scriptlet: initscripts-10.16-2.fc36.x86_64                        2/4
  Erasing          : chkconfig-1.19-2.fc36.x86_64                           3/4
  Erasing          : initscripts-rename-device-10.16-2.fc36.x86_64          4/4
  Running scriptlet: initscripts-rename-device-10.16-2.fc36.x86_64          4/4
  Verifying        : chkconfig-1.19-2.fc36.x86_64                           1/4
  Verifying        : initscripts-10.16-2.fc36.x86_64                        2/4
  Verifying        : initscripts-rename-device-10.16-2.fc36.x86_64          3/4
  Verifying        : wazuh-indexer-4.3.4-1.x86_64                           4/4

Removed:
  chkconfig-1.19-2.fc36.x86_64                  initscripts-10.16-2.fc36.x86_64
  initscripts-rename-device-10.16-2.fc36.x86_64 wazuh-indexer-4.3.4-1.x86_64

Complete!
28/06/2022 18:01:02 INFO: Wazuh indexer removed.
28/06/2022 18:01:02 INFO: Installation cleaned. Check the /var/log/wazuh-install.log file to learn more about the issue.
```

</details>

- openssl command not found

```
28/06/2022 18:00:01 INFO: Generating configuration files.
./wazuh-install.sh: line 2840: openssl: command not found
```

- The installation process continues and tries to create the tar file which contains the passwords and certificates
- Installation fails due to no certificates found in tar file
- The check function cert_checkOpenSSL seems to be not working
- The assistant should install openssl if not present in the system

```
function cert_checkOpenSSL() {
    if [ -z "$(command -v openssl)" ]; then
        common_logger -e "OpenSSL not installed."
        exit 1
    fi
}
```

```
[root@fedora36 vagrant]# openssl
bash: openssl: command not found
[root@fedora36 vagrant]# command -v openssl
[root@fedora36 vagrant]# 
[root@fedora36 vagrant]#
```


- After installing openssl manually, wazuh indexer was installed succesfully, but wazuh manager installation fails due to gpg check FAILED

```
Problem opening package wazuh-manager-4.3.4-1.x86_64.rpm
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'yum clean packages'.
Error: GPG check FAILED
```

- Fedora 36 is not listed as recommended system but we should see if the installation process has some type of error which could affect the recommended systems.

Regards, Raúl.

### PR DESCRIPTION
**Pull request by miguelfdez99**
|Related issue|
|---|
|closes #1702|

## Description

The error occurred because the function checks if the dependency openssl is installed with this command.
````
[root@fedora36 vagrant]# openssl
bash: openssl: command not found
[root@fedora36 vagrant]# yum list installed 2>/dev/null | grep openssl
openssl-devel.x86_64                 1:3.0.5-1.fc36                    @updates 
openssl-libs.x86_64                  1:3.0.5-1.fc36                    @updates 
````
The result is that the dependency is installed but actually it is not. We can not run OpenSSL unless openssl.x86_64 is installed.
That is why we need an extra check for the openssl dependency.
## Logs example

````
[root@fedora36 vagrant]# bash wazuh-install.sh -a -i
08/08/2022 09:04:44 INFO: Starting Wazuh installation assistant. Wazuh version: 4.3.6
08/08/2022 09:04:44 INFO: Verbose logging redirected to /var/log/wazuh-install.log
08/08/2022 09:04:45 WARNING: Hardware and system checks ignored.
08/08/2022 09:04:46 INFO: --- Dependencies ---
08/08/2022 09:04:46 INFO: Installing openssl.
08/08/2022 09:04:48 INFO: Wazuh repository added.
08/08/2022 09:04:48 INFO: --- Configuration files ---
08/08/2022 09:04:48 INFO: Generating configuration files.
08/08/2022 09:04:50 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
08/08/2022 09:04:50 INFO: --- Wazuh indexer ---
08/08/2022 09:04:50 INFO: Starting Wazuh indexer installation.
08/08/2022 09:05:39 INFO: Wazuh indexer installation finished.
08/08/2022 09:05:39 INFO: Wazuh indexer post-install configuration finished.
08/08/2022 09:05:39 INFO: Starting service wazuh-indexer.
08/08/2022 09:05:47 INFO: wazuh-indexer service started.
08/08/2022 09:05:47 INFO: Initializing Wazuh indexer cluster security settings.
08/08/2022 09:05:52 INFO: Wazuh indexer cluster initialized.
08/08/2022 09:05:52 INFO: --- Wazuh server ---
08/08/2022 09:05:52 INFO: Starting the Wazuh manager installation.
08/08/2022 09:06:23 INFO: Wazuh manager installation finished.
08/08/2022 09:06:23 INFO: Starting service wazuh-manager.
08/08/2022 09:06:34 INFO: wazuh-manager service started.
08/08/2022 09:06:34 INFO: Starting Filebeat installation.
08/08/2022 09:06:39 INFO: Filebeat installation finished.
08/08/2022 09:06:39 INFO: Filebeat post-install configuration finished.
08/08/2022 09:06:39 INFO: Starting service filebeat.
08/08/2022 09:06:40 INFO: filebeat service started.
08/08/2022 09:06:40 INFO: --- Wazuh dashboard ---
08/08/2022 09:06:40 INFO: Starting Wazuh dashboard installation.
08/08/2022 09:07:22 INFO: Wazuh dashboard installation finished.
08/08/2022 09:07:22 INFO: Wazuh dashboard post-install configuration finished.
08/08/2022 09:07:22 INFO: Starting service wazuh-dashboard.
08/08/2022 09:07:22 INFO: wazuh-dashboard service started.
08/08/2022 09:07:46 INFO: Initializing Wazuh dashboard web application.
08/08/2022 09:07:46 INFO: Wazuh dashboard web application initialized.
08/08/2022 09:07:46 INFO: --- Summary ---
08/08/2022 09:07:46 INFO: You can access the web interface https://<wazuh-dashboard-ip>
    User: admin
    Password: R.KXvtxRiy+j2J?hOWAl6gWUYi?n32aJ
08/08/2022 09:07:46 INFO: Installation finished.
````
````
[root@fedora36 vagrant]# yum list installed 2>/dev/null | grep openssl.x86
openssl.x86_64                       1:3.0.5-1.fc36                    @updates 
````

### CODE REVIEW

File: unattended_installer/install_functions/installCommon.sh [hunk 1]
```diff
@@ -266,7 +266,11 @@ function installCommon_installPrerequisites() {
         dependencies=( curl libcap tar gnupg openssl )
         not_installed=()
         for dep in "${dependencies[@]}"; do
-            if [ -z "$(yum list installed 2>/dev/null | grep ${dep})" ];then
+            if [ "${dep}" == "openssl" ]; then
+                if ! yum list installed 2>/dev/null | grep -q "${dep}".x86;then
```
Comment (okynos) [inline unattended_installer/install_functions/installCommon.sh:270]:
```suggestion
                if ! yum list installed 2>/dev/null | grep -q "${dep}\.";then
```

**Review by okynos — CHANGES_REQUESTED** [review]
Minor change

File: unattended_installer/install_functions/installCommon.sh (hunk 1 above)
Comment (miguelfdez99) [reply unattended_installer/install_functions/installCommon.sh]: Done https://github.com/wazuh/wazuh-packages/pull/1779/commits/fb652b2b32ffd46d1a284cb767dc187a6ded0cc2

File: unattended_installer/install_functions/installCommon.sh [hunk 2]
```diff
@@ -266,7 +266,11 @@ function installCommon_installPrerequisites() {
         dependencies=( curl libcap tar gnupg openssl )
         not_installed=()
         for dep in "${dependencies[@]}"; do
-            if ! yum list installed 2>/dev/null | grep -q "${dep}" ;then
+            if [ "${dep}" == "openssl" ]; then
+                if ! yum list installed 2>/dev/null | grep -q "${dep}\.";then
+                not_installed+=("${dep}")
```
Comment (DFolchA) [inline unattended_installer/install_functions/installCommon.sh:271]: Check indent

**Review by DFolchA — CHANGES_REQUESTED** [review]

File: unattended_installer/install_functions/installCommon.sh (hunk 2 above)
Comment (miguelfdez99) [reply unattended_installer/install_functions/installCommon.sh]: Done https://github.com/wazuh/wazuh-packages/pull/1779/commits/e7ac44ea17638aaa32e3e7979932c5d11bcb4b76

**Review by okynos — APPROVED** [review]

**Review by DFolchA — APPROVED** [review]
lgtm

**Pull request merged**